components/ActivityFilter/index.js

```diff
@@ -0,0 +1,35 @@
+import styled from "styled-components";
+import categories from "@/assets/categories";
+export default function ActivityFilter({ onFilterActivities, filter }) {
+  return (
+    <FilterContainer>
+      {categories.map((category) => (
+        <FilterButton
+          key={category.id}
+          onClick={() => onFilterActivities(category.id)}
+          style={{
+            backgroundColor: filter === category.id ? "orange" : "black",
+          }}
```

Comment:
```suggestion
          $filter
```
Oh, I almost forgot this line. Instead of setting the backgroundColor directly within the style prop, I recommend passing a prop to the FilterButton component to handle the styling conditionally:
```
const FilterButton = styled.button`
  padding: 5px;
  border-radius: 5px;
  ...
   background-color: ${(props) => (props.$filter ? "orange" : "black")};
`;
```